spec/features/users/user_can_create_a_binary_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.feature "User can create a new binary" do
+  context "As a logged in User" do
+    attr_reader :user
+
+    before :each do
+      @user = create(:user)
+    end
+
+    it "click on upload file and be redirected to correct path" do
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+      visit "/#{user.username}/home"
+      click_on "Upload File"
+
+      expect(current_path).to eq("/#{user.username}/home/binary_new")
+    end
+
+    xit "can uplaod a file" do
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+      visit "/#{user.username}/home/binary_new"
+
+      attach_file(:binary_data_url, Rails.root + "spec/fixtures/files/test.txt")
```

Comment:
@case-eee This is confusing us. When we upload a file through the browser, params returns the S3 url pointing to our object like we would expect. When we do it through the test, params returns an ActionDispatch object (Which contains the relevant info). Any idea what's going on?